cookbook/build-admin-frontend.rst

```diff
@@ -77,5 +77,31 @@ on your computer.
     npm install
     npm run build
 
+Common errors
+-------------
+
+If the installation of your dependencies or the webpack build itself fails, there are some quick fixes that may help you fix the problem:
+
+1. Use a supported Node.JS and NPM version
+
+Generally, the Node.JS and NPM versions used in Sulu's `Test Application workflow`_ are the only ones, which are officially supported by us.
+Those are Node.JS v12 and v14 and NPM v6. Newer versions might be working though.
+
+.. warning::
+
+    Sulu is currently not compatible with NPM v7, see `this issue`_.
+
+2. Build fails for no obvious reason
+
+It may happen, that the build fails for no obvious reason.
+In that case, it's a good idea to remove all the `package-lock.json` files and `node_modules` directories below your project root as described above.
+If that still doesn't help, you can try cleaning npm's internal cache by running:
+
+.. code-block:: bash
+
+    npm cache clean --force
```

Comment:
```suggestion
The webpack build might fail because of leftovers from previous builds our outdated packages.
To prevent this, you should remove all the `package-lock.json` files and `node_modules` directories below your project root before installing the npm dependencies:

.. code-block:: bash

   rm -rf assets/admin/node_modules && rm -rf vendor/sulu/sulu/node_modules && rm -rf vendor/sulu/sulu/src/Sulu/Bundle/*/Resources/js/node_modules
   rm -rf assets/admin/package-lock.json && rm -rf vendor/sulu/sulu/package-lock.json && rm -rf vendor/sulu/sulu/src/Sulu/Bundle/*/Resources/js/package-lock.json

If this does not solve the problem, you can try to clean the npm cache on your machine to prevent installing cached packages:

.. code-block:: bash

    npm cache clean --force
```